pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -1122,4 +1129,57 @@ func init() {
 	if err != nil {
 		panic(err)
 	}
+
+	err = InfraSAs.addServiceAccount(
+		InfraTemplateInstanceControllerServiceAccountName,
+		authorizationapi.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: TemplateInstanceControllerRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				{
+					APIGroups: []string{templateapi.GroupName},
+					// "impersonate" is required for the API server to accept updates to
+					// TemplateInstance objects where the requester username is not the
+					// API caller.
+					Verbs:     sets.NewString("get", "list", "watch", "update", "impersonate"),
+					Resources: sets.NewString("templateinstances"),
+				},
+				{
+					APIGroups: []string{kapi.GroupName},
+					Verbs:     sets.NewString("get"),
+					Resources: sets.NewString("secrets"),
+				},
+				{
+					APIGroups: []string{"*"},
+					// delete is needed to avoid error: "cannot set an ownerRef on a
+					// resource you can't delete"
+					Verbs:     sets.NewString("create", "delete"),
```

Comment:
This is the controller that actually does the creation of resources from the template?  He can't be allowed to have this permission.  This would allow me to create `oauthaccesstokens` which I could use to create a super-user token and own the cluster.

You will have to enumerate.